astroquery/gemini/core.py

```diff
@@ -419,6 +422,271 @@ def query_raw(self, *args, **kwargs):
         js = response.json()
         return _gemini_json_to_table(js)
 
+    @class_or_instance
+    def query_calibrations_for_region(self, coordinates, radius=0.3*units.deg):
+        """
+        search for Gemini calibrations for an ovservation query by target on the sky.
+
+        This call pairs with `query_region` and returns the calibrations for the
+        observations returned with a `query_region` search.
+
+        Parameters
+        ----------
+        coordinates : str or `~astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `~astropy.coordinates` object.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.3 degrees.
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used. Defaults to 0.3 deg.
+
+        Returns
+        -------
+        response : `~astropy.table.Table`
+        """
+        return self.query_calibrations_for_criteria(coordinates=coordinates, radius=radius)
+
+    @class_or_instance
+    def query_calibrations_for_object(self, objectname, radius=0.3*units.deg):
+        """
+        search for calibrations relating to the given object search.
+
+        This is the calibration query for the `query_object` search.  This will find the
+        calibrations relating to the observations for that searc.
+
+        Parameters
+        ----------
+        objectname : str
+            The name of an object to search observations for.  This attempts to resolve
+            the object by name and do a search on that area of the sky.  This does not handle
+            moving targets.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.3 degrees.
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used. Defaults to 0.3 deg.
+
+        Returns
+        -------
+        response : `~astropy.table.Table`
+        """
+        return self.query_calibrations_for_criteria(objectname=objectname, radius=radius)
+
+    @class_or_instance
+    def query_calibrations_for_criteria(self, *rawqueryargs, coordinates=None, radius=None, pi_name=None, program_id=None, utc_date=None,
```

Comment:
what are the possible rawqueryargs and rawquerykwargs possibilities? We preferably should not have those wildcard type options.

Reply:
The calls are really mapping to a URL search on the Gemini archive website.  These consist of an arbitrary set of arguments in the URL path or key=value arguments as well.  It's order independent on the website.

In the case of `query_raw` and `query_calibrations_raw` I really prefer this open-ended approach so the API mirrors what the web API is doing.  For the `_criteria` calls I could drop them.  They are just a convenience if you want to use the more well defined functions but still slip in one or two unusual criteria or terms.